### PR TITLE
Support new, generic $approx_set as a column statistics aggregation

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import io.trino.spi.type.StandardTypes;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.FAIL_ON_NULL;
+import static io.trino.spi.function.OperatorType.XX_HASH_64;
+import static io.trino.util.Failures.internalError;
+
+@AggregationFunction(value = "approx_set_generic", hidden = true) // approx_set_generic is hidden as used only for statistics collection in ANALYZE so far; eventually it may be used for implementation of user visible approx_set
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java:42]: Call it $approx_set. That way, it’s harder to call it inadvertently.

File: core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        hll.addHash(hash);
+        state.addMemoryUsage(hll.estimatedInMemorySize());
+    }
+
+    @InputFunction
+    public static void input(@AggregationState HyperLogLogState state, @SqlType("BOOLEAN") boolean value)
+    {
+        HyperLogLog hll = getOrCreateHyperLogLog(state);
+        state.addMemoryUsage(-hll.estimatedInMemorySize());
+        hll.add(value ? 1 : 2);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java:112]: nit is `addHash` faster? :)

File: core/trino-main/src/main/java/io/trino/sql/planner/StatisticsAggregationPlanner.java [hunk 3]
```diff
@@ -113,6 +114,8 @@ private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticType
                 return createAggregation(QualifiedName.of("max"), input.toSymbolReference(), inputType, inputType);
             case NUMBER_OF_DISTINCT_VALUES:
                 return createAggregation(QualifiedName.of("approx_distinct"), input.toSymbolReference(), inputType, BIGINT);
+            case NUMBER_OF_DISTINCT_VALUES_SUMMARY:
+                return createAggregation(QualifiedName.of("approx_set_generic"), input.toSymbolReference(), inputType, HYPER_LOG_LOG);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/StatisticsAggregationPlanner.java:118]: update the commit message

> Support approx_set as a column statistics aggregation

or add a commit description explaining why it's not the actual
approx_set what is being used.

**Review by findepi — APPROVED** [review]
% naming per Martin's

File: core/trino-main/src/main/java/io/trino/sql/planner/StatisticsAggregationPlanner.java (hunk 3 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/sql/planner/StatisticsAggregationPlanner.java:118]: I updated the commit message and added a comment explaining why `$approx_set` is used

File: core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java (hunk 2 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java:112]: Yeah - should be. We do not need to compute hash. Changed to use addHash with precomputed hash values.

**Pull request merged**